app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/DeviceListFragment.java

```diff
@@ -0,0 +1,357 @@
+ * 2) After handshake, starting the files transfer using {@link SenderDeviceAsyncTask} on the sender
+ *    device and {@link ReceiverDeviceAsyncTask} files receiving device
+ *
+ * The starting point for the module is {@link LocalFileTransferActivity}
+ * */
+public class DeviceListFragment extends ListFragment implements WifiP2pManager.PeerListListener, WifiP2pManager.ConnectionInfoListener {
+
+  public static final String TAG = "DeviceListFragment";
+  public static int PEER_HANDSHAKE_PORT = 8009;
+  public static int FILE_TRANSFER_PORT = 8008;
+
+  private SharedPreferenceUtil sharedPreferenceUtil;
+
+  private LocalFileTransferActivity localFileTransferActivity;  // Parent activity, starting point of the module
+  private TransferProgressFragment transferProgressFragment;    // Sibling fragment, for displaying transfer progress
+
+  private View deviceListFragmentRootView = null; // Root view corresponding to the DeviceListFragment
+
+  private boolean fileSender = false; // Whether file sending device or not
+  private ArrayList<Uri> fileUriList; // Uris of files to be shared, available only for the sender device
+  private int totalFilesForTransfer = -1;
+  private int filesSent = 0;          // Count of number of files transferred until now
+  private ArrayList<FileItem> filesToSend = new ArrayList<>();
+
+  private WifiP2pDevice userDevice;   // Represents the device on which the app is running
+  private WifiP2pInfo groupInfo;      // Corresponds to the WiFi P2P group formed between the two devices
+  private List<WifiP2pDevice> peerDevices = new ArrayList<WifiP2pDevice>();
+
+  private WifiP2pDevice selectedPeerDevice = null;
+  private InetAddress selectedPeerDeviceInetAddress;
+
+  private InetAddress fileReceiverDeviceAddress;  // IP address of the file receiving device
+  private boolean fileTransferStarted = false;
+
+  @Override
+  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+    super.onActivityCreated(savedInstanceState);
+    localFileTransferActivity = (LocalFileTransferActivity) getActivity();
+
+    // As DeviceListFragment extends ListFragment for the purpose of displaying list of peers
+    this.setListAdapter(new WifiPeerListAdapter(localFileTransferActivity, R.layout.row_peer_device, peerDevices));
+
+    if(localFileTransferActivity.isFileSender()) {
+      fileSender = true;
+      fileUriList = localFileTransferActivity.getFileUriArrayList();
+      totalFilesForTransfer = fileUriList.size();
+
+      for(int i = 0; i < fileUriList.size(); i++) {
+        filesToSend.add(new FileItem(getFileName(fileUriList.get(i)), FileItem.TO_BE_SENT));
+      }
+
+      displayTransferProgressFragment();
+    }
+
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    deviceListFragmentRootView = inflater.inflate(R.layout.fragment_device_list, null);
```

Comment:
this does not need to be assigned, all fragments have the getView method to return their inflated view.
If you need to find views then do so with butterknife, or even more preferably with kotlin synthetic properties

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/28840898e008c346ac4b5b562eec5026cae3c28d#diff-c2e372482ef0816a999e44dec2c63c52) using butterknife